doc/release/upcoming_changes/25347.c_api.rst

```diff
@@ -2,8 +2,5 @@
   exposed. This includes functions for acquiring and releasing mutexes locking
   access to the string data as well as packing and unpacking UTF-8 bytestreams
   from array entries.
-* ``NPY_NTYPES`` is now version-dependent to accomadate the availability of a
-  new NumPy built-in DType. It is now defined in ``npy2_compat.h``.
-* ``NPY_NTYPES_LEGACY`` is now defined in ``ndarraytypes.h``, use this value
-  if you do not wish to update code to support dtypes written using the new
-  DType API that may not function in the same way as legacy dtypes.
+* ``NPY_NTYPES`` has been renamed to ``NPY_NTYPES_LEGACY`` as it does not
+  include new NumPy built-in DTypes (specially the string DType).
```

Comment:
```suggestion
  include new NumPy built-in DTypes. In particular the new string DType
  will likely not work correctly with code that handles legacy DTypes.
```

Would it maybe be useful for downstream to have a `NTYPES_NONLEGACY` that is set to 1 now? I guess we can always add that later.

Reply:
Yah, let's add it later, right now I am not even sure that `NPY_NTYPES` has a single user.